packages/builder/src/builder.js

```diff
@@ -437,15 +434,20 @@ export default class Builder {
       ...templateContext.templateFiles
     ])
 
+    const appDir = path.resolve(this.options.srcDir, this.options.dir.app)
+
     templateContext.templateFiles = await Promise.all(templatePaths.map(async (file) => {
       // Use custom file if provided in build.templates[]
       const customTemplateIndex = customTemplateFiles.indexOf(file)
       const customTemplate = customTemplateIndex !== -1 ? this.options.build.templates[customTemplateIndex] : null
       let src = customTemplate ? (customTemplate.src || customTemplate) : r(this.template.dir, file)
+
       // Allow override templates using a file with same name in ${srcDir}/app
-      const customPath = r(this.options.srcDir, this.options.dir.app, file)
-      const customFileExists = await fsExtra.exists(customPath)
-      src = customFileExists ? customPath : src
+      const customPath = path.resolve(this.options.srcDir, this.options.dir.app, file)
+      const customFileExists = customPath.indexOf(appDir) === 0 && await fsExtra.exists(customPath)
```

Comment:
The real fix is actually here: `customPath.indexOf(appDir) === 0`

Reply:
I changes `r` to `path.resolve` as aliases are meaningless in this context. We resolve relative to appDir.

Reply:
Using `startsWith` might be just a bit more readable?